BTL TCP compile error on Cisco MTT

master-201810300304-9c44e8c, does not appear to be on other branches. 

configure line:
```
"CFLAGS=-g -pipe" --enable-picky --enable-debug --enable-mpirun-prefix-by-default --enable-mpi-cxx --disable-dlopen --enable-ipv6
```

stderr:
```
--- make all -j 32 result_stderr ---
keyval_lex.c: In function ‘yy_get_next_buffer’:
keyval_lex.c:751:18: warning: comparison between signed and unsigned integer expressions
[-Wsign-compare]
   for ( n = 0; n < max_size && \
                  ^
keyval_lex.c:1284:3: note: in expansion of macro ‘YY_INPUT’
   YY_INPUT( (&YY_CURRENT_BUFFER_LVALUE->yy_ch_buf[number_to_move]),
   ^~~~~~~~show_help_lex.c: In function ‘yy_get_next_buffer’:
show_help_lex.c:647:18: warning: comparison between signed and unsigned integer expressions
[-Wsign-compare]
   for ( n = 0; n < max_size && \
                  ^
show_help_lex.c:1081:3: note: in expansion of macro ‘YY_INPUT’
   YY_INPUT( (&YY_CURRENT_BUFFER_LVALUE->yy_ch_buf[number_to_move]),
   ^~~~~~~~btl_openib_lex.c: In function ‘yy_get_next_buffer’:
btl_openib_lex.c:703:18: warning: comparison between signed and unsigned integer expressions
[-Wsign-compare]
   for ( n = 0; n < max_size && \
                  ^
btl_openib_lex.c:1226:3: note: in expansion of macro ‘YY_INPUT’
   YY_INPUT( (&YY_CURRENT_BUFFER_LVALUE->yy_ch_buf[number_to_move]),
   ^~~~~~~~In file included from
/home/mpiteam/scratches/community/2018-10-29cron/WoeO/mpi-install/CTmK/src/openmpi-master-201810300304-9c44e8c/opal/mca/hwloc/hwloc201/hwloc/include/hwloc/bitmap.h:17:0,
                 from ../../../../opal/mca/hwloc/hwloc201/hwloc/include/hwloc.h:71,
                 from ../../../../opal/mca/hwloc/hwloc201/hwloc201.h:26,
                 from ../../../../opal/mca/hwloc/hwloc-internal.h:131,
                 from ../../../../opal/util/proc.h:22,
                 from btl_tcp_endpoint.c:57:
btl_tcp_endpoint.c: In function ‘mca_btl_tcp_endpoint_start_connect’:
btl_tcp_endpoint.c:745:53: error: ‘struct mca_btl_tcp_module_t’ has no member named
‘tcp_ifaddr_6’; did you mean ‘tcp_ifaddr’?
         assert(NULL != &btl_endpoint->endpoint_btl->tcp_ifaddr_6);
                                                     ^make[2]: *** [btl_tcp_endpoint.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all-recursive] Error 1
make: *** [all-recursive] Error 1
```

--- a/opal/mca/btl/tcp/btl_tcp_endpoint.c
+++ b/opal/mca/btl/tcp/btl_tcp_endpoint.c
@@ -742,7 +742,7 @@ static int mca_btl_tcp_endpoint_start_connect(mca_btl_base_endpoint_t* btl_endpo
     }
 #if OPAL_ENABLE_IPV6
     if (endpoint_addr.ss_family == AF_INET6) {
-        assert(NULL != &btl_endpoint->endpoint_btl->tcp_ifaddr_6);
+        assert(NULL != &btl_endpoint->endpoint_btl->tcp_ifaddr);
         if (bind(btl_endpoint->endpoint_sd, (struct sockaddr*) &btl_endpoint->endpoint_btl->tcp_ifaddr,
                  sizeof(struct sockaddr_in6)) < 0) {
             BTL_ERROR(("bind on local address (%s:%d) failed: %s (%d)",

